lightning/src/util/wakers.rs

```diff
@@ -112,18 +120,23 @@ pub(crate) struct FutureState {
 	// When we're tracking whether a callback counts as having woken the user's code, we check the
 	// first bool - set to false if we're just calling a Waker, and true if we're calling an actual
 	// user-provided function.
```

Comment:
nit: Maybe we can update the comment to make it more general than talking specifically about `first bool`?